### PR TITLE
feat: [CDS-37792]: Feature for searching users in harness-ng

### PR DESCRIPTION
**Pull request by mitishapandey**
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35671)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Code issue 7 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35671)

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 13 files at r1, 1 of 8 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35671)* status: 13 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/jira/resources/service/JiraResourceServiceImpl.java` line 129 at r4](https://reviewable.io/reviews/harness/harness-core/35671#-N8ZoQdSBeOrtqhQ-0Tn:-N8ZoQdSBeOrtqhQ-0To:b555i80) ([raw file](https://github.com/harness/harness-core/blob/a190c58a4df070095e843559271a5f7df292eb49/125-cd-nextgen/src/main/java/io/harness/cdng/jira/resources/service/JiraResourceServiceImpl.java#L129)):*
> ```Java
>       String connectorId, long defaultSyncCallTimeout, String userQuery, String offset) {
>     if (!cdFeatureFlagHelper.isEnabled(accountId, FeatureName.ALLOW_USER_TYPE_FIELDS_JIRA)) {
>       return null;
> ```

add null checks
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/jira/JiraTaskNGHandler.java` line 85 at r4](https://reviewable.io/reviews/harness/harness-core/35671#-N8Zmf-WA9LvNa0qlODZ:-N8Zmf-WA9LvNa0qlOD_:bglnkmz) ([raw file](https://github.com/harness/harness-core/blob/a81004e81d8b14d430cb10e4b7f3eab321550517/930-delegate-tasks/src/main/java/io/harness/delegate/task/jira/JiraTaskNGHandler.java#L85)):*
> ```Java
>     Set<JiraFieldNG> jiraUserFields = new HashSet<>();
>     createMetadata.getProjects().values().forEach(proj
>         -> proj.getIssueTypes().values().forEach(issueType -> issueType.getFields().values().stream().forEach(field -> {
> ```

wrap this behind FF instead of complete removal 
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/jira/JiraTaskNGHandler.java` line 108 at r4](https://reviewable.io/reviews/harness/harness-core/35671#-N8ZoAO-2NH68jqL3OlE:-N8ZoAO-2NH68jqL3OlF:b-e37seg) ([raw file](https://github.com/harness/harness-core/blob/a190c58a4df070095e843559271a5f7df292eb49/930-delegate-tasks/src/main/java/io/harness/delegate/task/jira/JiraTaskNGHandler.java#L108)):*
> ```Java
>       JiraIssueTypeNG issueType = project.getIssueTypes().get(params.getIssueType());
>       issueType.getFields().entrySet().forEach(e -> {
>         if (e.getValue().getSchema().getType().equals(JiraFieldTypeNG.USER)) {
> ```

this should be a method as same code is replicated in updateissue



_Code quote:_
```Java
      issueType.getFields().entrySet().forEach(e -> {
        if (e.getValue().getSchema().getType().equals(JiraFieldTypeNG.USER)) {
          userTypeFields.add(e.getKey());
        }
      });
```

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java [hunk 1]
```diff
@@ -233,6 +248,137 @@ public void testgetIssue() throws Exception {
...
+    assertThat(jiraTaskNGResponse.getIssue()).isNotNull();
+    assertThat(jiraTaskNGResponse).isNotNull();
+    assertThat(jiraTaskNGResponse.getIssue()).isEqualTo(jiraIssueNG);
+    assertThat(jiraTaskNGParameters.getFields()).isEqualTo(fields1);
+  }
+
+  @Test
+  @Owner(developers = YUVRAJ)
+  @Category(UnitTests.class)
+  public void testupdateIssue() throws Exception {
```
Comment (lasdox) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java:322]: please add a test for when jira.getUsers returns more than 1 user (it should throw exception)

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGResponse.java [hunk 2]
```diff
@@ -37,10 +37,12 @@
 public class JiraTaskNGResponse implements DelegateTaskNotifyResponseData {
   List<JiraProjectBasicNG> projects;
   List<JiraStatusNG> statuses;
+  // List<JiraUserData> jiraUserDataList;
```
Comment (lasdox) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGResponse.java:40]: remove this comment

**Review by lasdox — DISMISSED** [review]
> Reviewed 5 of 13 files at r1, 1 of 5 files at r3, 2 of 8 files at r4, 1 of 2 files at r5, 6 of 6 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35671)* status: :shipit: complete! ✓

**Comment by lasdox** [comment]
Please make all sanity testing in PR env before merging this

**Comment by Y0709** [comment]
trigger ti0, ti1, ti2, ti3,ti4

**Comment by Y0709** [comment]
trigger ti3

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35671)* status: 8 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/jira/resources/service/JiraResourceServiceImpl.java` line 123 at r9](https://reviewable.io/reviews/harness/harness-core/35671#-NAyo8V_-gS3nWZ-EaOQ:-NAyo8V_-gS3nWZ-EaOR:b545a63) ([raw file](https://github.com/harness/harness-core/blob/2a7c6fb0a0fe4e51cd3cd4061dd2bb4af858001b/125-cd-nextgen/src/main/java/io/harness/cdng/jira/resources/service/JiraResourceServiceImpl.java#L123)):*
> ```Java
>       Set<JiraFieldNG> jiraUserFields = new HashSet<>();
>       jiraIssueCreateMetadataNG.getProjects().values().forEach(
>           proj -> proj.getIssueTypes().values().forEach(issueType1 -> issueType1.getFields().values().forEach(field -> {
> ```

so many for each?
any way to simplify?

**Comment by Y0709** [comment]
trigger codebasehash

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 4 of 13 files at r1, 1 of 5 files at r3, 1 of 8 files at r4, 2 of 6 files at r6, 1 of 4 files at r7, 2 of 2 files at r8, 5 of 5 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35671)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35671)* status: :shipit: complete! ✓

**Comment by abhinav-harness** [comment]
trigger codebasehash

**Pull request merged**